internal/controller/processor.go

```diff
@@ -75,7 +81,44 @@ func (processor *AppConfigurationProviderProcessor) ProcessFullReconciliation()
 	return nil
 }
 
-func (processor *AppConfigurationProviderProcessor) ProcessKeyValueRefresh() error {
+func (processor *AppConfigurationProviderProcessor) ProcessFeatureFlagRefresh(existingConfigMap *corev1.ConfigMap) error {
+	provider := *processor.Provider
+	reconcileState := processor.ReconciliationState[processor.NamespacedName]
+	currentTime := processor.CurrentTime
+	var err error
+	// Check if the feature flag dynamic feature if enabled
+	if provider.Spec.FeatureFlag != nil && provider.Spec.FeatureFlag.Refresh != nil && provider.Spec.FeatureFlag.Refresh.Enabled {
+		processor.RefreshOptions.featureFlagRefreshEnabled = true
+	} else {
+		reconcileState.NextFeatureFlagRefreshReconcileTime = metav1.Time{}
+		return nil
+	}
+
+	refreshInterval, _ := time.ParseDuration(provider.Spec.FeatureFlag.Refresh.Interval)
+	nextFeatureFlagRefreshReconcileTime := metav1.Time{Time: currentTime.Add(refreshInterval)}
+	if processor.ShouldReconcile {
+		reconcileState.NextFeatureFlagRefreshReconcileTime = nextFeatureFlagRefreshReconcileTime
+		return nil
+	}
+
+	if !currentTime.After(reconcileState.NextFeatureFlagRefreshReconcileTime.Time) {
+		return nil
+	}
+
+	processor.RefreshOptions.featureFlagRefreshNeeded = true
+	partiallyUpdatedSettings, err := (*processor.Retriever).RefreshFeatureFlagSettings(processor.Context, &existingConfigMap.Data)
```

Comment:
Do you mind rename `partiallyUpdatedSettings` to `featureFlagRefreshedSettings`?